lib/demangler/base.js

```diff
@@ -162,21 +164,41 @@ export class BaseDemangler extends AsmRegex {
 
         const translations = [...this.symbolstore.listTranslations(), ...this.othersymbols.listTranslations()]
             .filter(elem => elem[0] !== elem[1]);
+        translations.sort();
```

Comment:
So the reason this was not sorted alphabetically, is because it's sorted by length. This to make sure that if you have partially duplicate symbols, they don't get mistaken for the other.

So things like:
* zabc
* abc

If you replace abc first, then it can't demangle the original.

Reply:
Aha! Got it. I think I can recover that by finding the _longest_ match, which might actually already be happening because the way the upper_bound works.

That said my brain is hurting now, can you think is of a specific example? I'm amazed none of the `bug-*` picked it up (I created the "expected" output for bug-1336.asm with the original code)

Reply:
I don't think we ever made an example, can't find the issue in the closed issues with a quick glance either.
Will have more time to have a look and see tomorrow

Reply:
Thanks @partouf . I'll look too. I _think_ we might be ok because of the aforementioned "upper bound will find the longest prefix" _and_ we start from the beginning of the string and work forwards only. we don't look arbitrarily far ahead until we found the longest match at position 0, then position 1, etc.

But...this is super subtle and so I would appreciate your further thoughts on this! Thanks, look forward to hearing back when you have time.

Reply:
So I was wrong; and have reimplemented using a prefix tree! Thank you _so_ much for saving us from a potential catastophe!